File: PERL/Cap07/c07ex04.pl
# c07ex04.pl

#!/usr/bin/perl

use strict;
use warnings;

sub FATORIAL {
  my ($N) = @_;
  my $FAT = 1;
  for (my $I = 1; $I <= $N; $I++) {
    $FAT = $FAT * $I;
  }
  print "Fatorial = $FAT\n";
}

print "Qual fatorial: ";
my $LIMITE = <STDIN>;
chomp $LIMITE;

FATORIAL($LIMITE);

print "\n";
print "Tecle <Enter> para encerrar... ";
<STDIN>;

File: GO/Cap07/c07ex04.go
// c07ex04.go

package main
import "fmt"

func fatorial(N int) {
  var Fat, I int
  Fat = 1
  for I = 1; I <= N; I++ {
    Fat *= I
  }
  fmt.Printf("Fatorial = %d\n", Fat)
}

func main() {
  var Limite int

  fmt.Print("Qual fatorial: ")
  fmt.Scan(&Limite)

  fatorial(Limite)

  fmt.Println()
  fmt.Print("Tecle <Enter> para encerrar... ")
  fmt.Scanln()
}
